### PR TITLE
Creation of the timeline view

### PR DESCRIPTION
**Pull request by FuriousLlama**
[AB#7450](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/7450)
Created the view with the matching styling.
Created a timeline view model

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/app/models/medicationStatement.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  // Medication statement dosage.
+  public dosage!: number;
+  // Medication statement prescription status.
+  public prescriptionStatus!: string;
+  // Date the medication statement was dispensed.
+  public dispensedDate!: Date;
+  // Surname of the Practitioner who issued the medication statement.
+  public practitionerSurname!: string;
+  // Drug medication discontinued date, if applicable.
+  public drugDiscontinuedDate!: Date;
```
Comment (TiagoGraf89) [inline Apps/WebClient/src/ClientApp/app/models/medicationStatement.ts]: this is a nullable field (change from ! to ?)

File: Apps/WebClient/src/ClientApp/app/models/medicationStatement.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  // Medication statement directions.
+  public directions!: string;
+  // Date the medication statement was entered.
+  public dateEntered!: Date;
```
Comment (TiagoGraf89) [inline Apps/WebClient/src/ClientApp/app/models/medicationStatement.ts]: same here

File: Apps/WebClient/src/ClientApp/app/models/timelineEntry.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+export default class TimelineEntry {
+  public readonly date: Date;
+  public readonly id: string;
+  public readonly title: string;
+  public readonly description: string;
+  public readonly type: EntryType;
+
+  public constructor(other: MedicationStatement) {
+    // The ID needs to come from the the server
+    this.id = "id-" + Math.random();
```
Comment (TiagoGraf89) [inline Apps/WebClient/src/ClientApp/app/models/timelineEntry.ts]: lets move this to the medication service?

**Comment by healthgatewaybc** [comment]
SonarQube Report: https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=webclient_pr_118

File: Apps/WebClient/src/ClientApp/app/models/timelineEntry.ts (hunk 3 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/app/models/timelineEntry.ts]: Correct. However doing that will trickle all the way down to the parsing of the trf message. Probably another task

File: Apps/WebClient/src/ClientApp/app/models/medicationStatement.ts (hunk 1 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/app/models/medicationStatement.ts]: changed

File: Apps/WebClient/src/ClientApp/app/models/medicationStatement.ts (hunk 2 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/app/models/medicationStatement.ts]: changed

**Pull request merged**